streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -160,26 +172,190 @@ public void shouldIdempotentlyUpdateSubscriptionFromActiveAssignment() {
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, 15L));
+
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        writeCheckpointFile(taskId00, changelogOffsets);
+
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
+    }
+
+    @Test
+    public void shouldNotReportOffsetSumsForUnlockableTask() throws IOException {
+        expectLockFailedFor(taskId00);
+        makeTaskFolders("0_0");
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertTrue(taskManager.getTaskOffsetSums().isEmpty());
+    }
+
+    @Test
+    public void shouldNotReportOffsetSumsForUnassignedTaskWithoutCheckpoint() throws IOException {
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        expect(stateDirectory.checkpointFileFor(taskId00)).andReturn(getCheckpointFile(taskId00));
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
+
+        assertTrue(taskManager.getTaskOffsetSums().isEmpty());
+    }
+
+    @Test
+    public void shouldSkipInvalidOffsetsWhenComputingOffsetSum() throws IOException {
+        final Map<TopicPartition, Long> changelogOffsets = mkMap(
+            mkEntry(new TopicPartition("changelog", 1), -1L)
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, 0L));
+
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        writeCheckpointFile(taskId00, changelogOffsets);
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
 
-        replay(activeTaskCreator, stateDirectory);
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
+    }
 
-        final Map<TaskId, Long> taskOffsetSums = taskManager.getTaskOffsetSums();
+    @Test
+    public void shouldOffsetSumOverflowToLongMaxValue() throws IOException {
+        final long largeOffset = Long.MAX_VALUE / 2;
+        final Map<TopicPartition, Long> changelogOffsets = mkMap(
+            mkEntry(new TopicPartition("changelog", 1), largeOffset),
+            mkEntry(new TopicPartition("changelog", 2), largeOffset),
+            mkEntry(new TopicPartition("changelog", 3), largeOffset)
+        );
+        final Map<TaskId, Long> expectedOffsetSums = mkMap(mkEntry(taskId00, Long.MAX_VALUE));
 
-        verify(activeTaskCreator, stateDirectory);
+        expectLockObtainedFor(taskId00);
+        makeTaskFolders("0_0");
+        writeCheckpointFile(taskId00, changelogOffsets);
+        replay(stateDirectory);
+        taskManager.handleRebalanceStart(singleton("topic"));
 
-        assertThat(taskOffsetSums.keySet(), equalTo(mkSet(taskId01, taskId02, new TaskId(1, 1))));
+        assertThat(taskManager.getTaskOffsetSums(), is(expectedOffsetSums));
     }
 
```

Comment:
req: Could add a test for an offset that is negative but not `-1`? And another test with multiple offsets of different types, i.e., negative, unknown, etc. The latter could be the composite case, we already talked about.

Reply:
Sure, I'll add one based on the current code and can update it if we decide to allow overflow